platforms/extensibility/test-kit/src/integTest/groovy/org/gradle/testkit/runner/GradleRunnerIsolationIntegrationTest.groovy

```diff
@@ -53,7 +53,8 @@ class GradleRunnerIsolationIntegrationTest extends BaseGradleRunnerIntegrationTe
             .build()
 
         when:
-        buildScript """
+        buildFile.clear()
```

Comment:
💅 Did you consider introducing a single operation to reset the file: `buildFile.reset """ ...`? I guess the problem might be supporting script specific highlighting, right? If that's a concern I think I'd still appreciate something like `buildFileReset """....` more than two operations.

Reply:
Introducing a single operation for reset and write is effectively what `buildScript` function already did. The reason we decided to remove the function was to avoid the need to make a choice when writing tests and have fewer functions to remember overall. The choice could have been simplified by giving the function a better name, but the team felt there was no need for a separate function because the occasions to reset are so rare.

Overall, having a single flavor of functions with the same semantics seemed like a better option. Especially so, because the "append" semantic combines well with "resetting" the file first.